src/components/Login/LoginContainer.tsx

```diff
@@ -91,6 +99,59 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setLoginStage("SCAN");
   };
 
+  const shouldResendAlert = async (): Promise<boolean> => {
+    return new Promise(resolve => {
+      Alert.alert(
+        "Resend OTP?",
+        lastResendWarningMessage,
+        [
+          {
+            text: "CANCEL",
+            onPress: () => resolve(false)
+          },
+          {
+            text: "RESEND",
+            onPress: () => resolve(true)
+          }
+        ],
+        { cancelable: false }
+      );
+    });
+  };
+
+  const handleRequestOTP = async (fullNumber?: string): Promise<boolean> => {
+    try {
+      fullNumber = fullNumber ?? mobileNumber;
+      if (lastResendWarningMessage) {
+        if (!(await shouldResendAlert())) return false;
+      }
+      const response = await requestOTP(fullNumber, codeKey, endpointTemp);
+      lastResendWarningMessage = response.warning ?? "";
+      return true;
+    } catch (e) {
+      if (e instanceof LoginError) {
+        Alert.alert(
+          "Error",
+          e.message,
+          [
+            {
+              text: "OK",
+              onPress: () => {
+                if (e instanceof LoginLockedOutError) resetStage();
```

Comment:
Hmm, wonder if it's better to move this out so we treat `LoginError` and `LoginLockedOutError` as siblings. This provides flexibility when we want to add other types of LoginErrors that may not have the same format as this alert where there's only 1 OK button

Reply:
i think if there are other types of LoginErrors we can just capture them separately?

Reply:
Yep, why is `LoginLockedOutError` a special case then? Can it be treated in a block on its own?

Reply:
hmm...for the current alerts i think treating it in its own block is a bit verbose. but once we change to using the alert modals it won't be that much of an issue. will change it here then!

Reply:
Okay!

Reply:
will update the alerts to use the alert modals!